### PR TITLE
Initial implementation of subtabs in the site management panel

### PR DESCRIPTION
**Pull request by fushar**
Related to:

- https://github.com/Automattic/wp-calypso/issues/95385

## Proposed Changes

This is an initial implementation of hosting dashboard subtabs. The goal is to ship the subtab infrastructure first, so that we can build on top of it for each tab in parallel.

* Moved (renamed) `client/hosting/sites` to `client/sites`, which contains the site dashboard "infractructure".
  * My idea is that when we later move a tab to the new IA structure, we do it by:
    * moving it from `client/hosting/*` to `client/sites/*`, and
    * updating the URL (e.g. `/sites/marketing/:slug`)
    * Let me know if this refactor does no make sense or you have strong objection.
* Added capability for a panel to have sidebar containing subtabs.
* Implemented a dummy Marketing tab containing two subtabs. It's under the `untangling/hosting-menu` flag, which is set to true in dev and wpcalypso envs.

image

## Why are these changes being made?

We want to migrate the pages from the Hosting menu in the left sidebar, to the "global" site-specific hosting dashboard panels.

## Testing Instructions

1. Use the provided Calypso link.
2. Go to /sites then click any site.
3. Verify you can see the new `Marketing` tab.
4. Click that tab.
5. Verify you can see the subtabs.
6. Click around the subtabs. Verify that the URL changes.
7. Click the second subtab (Business Tools) then refresh the page. Verify that the URL works for a fresh page load.

> [!WARNING]  
> The mobile view design is still not implemented in this PR; will follow-up later. Including more polishes to the Desktop version.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by lsl — APPROVED** [review]
Generally LGTM, aware of the approach etc.

Concerns:
- Does the rename need to happen along with the addition of new pages? Not a big deal, it was just hard to follow what's new vs what's a move.
- I know they make a good first issue for this structure but do we really need to keep these marketing/business pages?

Nit, the alignment here seems a bit off:

Screenshot(91)

**Comment by fushar** [comment]
Thanks for the review!

> Does the rename need to happen along with the addition of new pages? Not a big deal, it was just hard to follow what's new vs what's a move.

I thought of associating the move with the PR which actually shows the intent of the move. But now I realized that it's not that practical, especially now we know that some things will be worked on by multiple people/teams (e.g. the Marketing tab, or the /sites CSS issues). So it also makes sense to move first; happy either way.

The end goal here is to have the components in its final location (i.e. not in `/hosting/*` or `/my-sites/*`), regardless of whether we want to release them via feature flag or all at once. It's so that we don't need to do the cleanup at the last step.

> I know they make a good first issue for this structure but do we really need to keep these marketing/business pages?

The choice of the example tab was arbitrary; it's just so that there is one tab to show off. Regardless, I believe we still need to view the Marketing tab. The default plan is to migrate whatever the relevant team revamps in the original Tools -> Marketing; we're just "untangling" them to the new tab. We should discuss more about this.

> Nit, the alignment here seems a bit off:

It's tricky on very wide screens. Looks like the panel content has fixed width 😅 so it's happening to the other tabs as well. I'll leave it as future PR; there's still much to polish for this sidebar.
